src/libraries/System.Private.CoreLib/src/System/Type.Enum.cs

```diff
@@ -124,7 +124,7 @@ private void GetEnumData(out string[] enumNames, out Array enumValues)
             // Insertion Sort these values in ascending order.
             // We use this O(n^2) algorithm, but it turns out that most of the time the elements are already in sorted order and
             // the common case performance will be faster than quick sorting this.
-            IComparer comparer = Comparer<object>.Default;
+            IComparer<object> comparer = Comparer<object>.Default;
```

Comment:
Would a better change be:
```C#
IComparer comparer = Comparer.Default;
```
?

Reply:
IComparer is not very often use interface in modern apps so I'd rather use IComparer<object> to help also with trimming.

Reply:
> IComparer is not very often use interface in modern apps so I'd rather use IComparer to help also with trimming.

`Comparer<object>.Default` returns an `ObjectComparer<object>`, which not only derives from `Comparer<T>` and implements `IComparer`, its Compare method does `Comparer.Default.Compare(x, y);` (and `Comparer.Default` returns `IComparer`).  You're saying the linker can trim out `IComparer` in that case?

Reply:
> You're saying the linker can trim out IComparer in that case?

Yes, because it tracks if there is any dispatch over the interface method.

> we're actually paying more at run time to use the generic here.

Moved away from the interface dispatch completely in this case.